### ISSUE TITLE
[Google3 Bug]: [A11y] Announcing button label twice on android app when talkback in ON

### ISSUE THREAD

**Issue by gargnikhil0763** [flutter/flutter#175325]
### Help us understand the severity of this issue

- [ ] causing severe production issues e.g. malfunctions or data loss
- [x] blocking next binary release
- [ ] blocking a client feature launch within a quarter
- [ ] nice-to-have but does not block a launch within the next quarter

### Steps to reproduce

1. Click a button (B1) on Page P1 which pushes Page P2 on top.
2. Dismiss the Page P2 by pressing back button.
3. Focus comes back to page P1.

### Expected results

Label on button B1 should be announced once by talkback.

### Actual results

 Label on button B1 is announced twice by talkback.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: false),
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: Column(
          children: [
            Text(
              'Button tapped $_counter time${_counter == 1 ? '' : 's'}.',
              key: Key('CountText'),
            ),
            FilledButton(onPressed: onPressed, child: Text('New Page')),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }

  void onPressed() async {
    await Navigator.push(
      context,
      MaterialPageRoute(builder: (context) => NewPage()),
    );
  }
}

class NewPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('New Page')),
      body: Center(child: Text('This is a new page.')),
    );
  }
}

```

</details>

### Screenshots or Video

https://github.com/user-attachments/assets/4bd2cc95-1008-4796-9b84-a9a017bc15a3

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[        ] [✓] Flutter (Channel google3, on Debian GNU/Linux rodete 6.12.32-1rodete1-amd64, locale en_US.UTF-8) [0ms]
[        ]     • Framework revision 75a77fec8c (88 days ago), 2025-06-18T00:00:00.000
[        ]     • Engine revision 75a77fec8c
[        ]     • Dart version b32559bca3
[        ] [✓] Android toolchain - develop for Android devices (Android SDK version Stable) [88ms]
[        ]     • Android SDK at google3
[        ]     • Emulator version unknown
[        ]     • Platform Stable, build-tools Stable
[        ]     • Java binary at: /usr/local/buildtools/java/jdk/bin/java
[        ]       This JDK was found in the system PATH.
[        ]       To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
[        ]     • Java version OpenJDK Runtime Environment (build 24.0.2+-google-release-sts-785595728)
[        ] [✓] VS Code (version unknown) [87ms]
[        ]     • VS Code at /usr/share/code
[        ]     • Flutter extension can be installed from:
[        ]       🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[        ]     ✗ Unable to determine VS Code version.
```

</details>

**Comment by tirth-patel-nc** [comment]
Thanks for the report. Seeing the same behaviour with latest SDK versions. Tested on Pixel 6a.

on 3.35.3, the button label is announced twice, but only for the on-screen bottom navigation bar’s back button and not for app bar's back button.

```
stable : 3.35.3
master : 3.37.0-1.0.pre-111
```

**Comment by chunhtai** [comment]
cc @hannah-hyj , I think this may due to the refocus when navigating back from another page
